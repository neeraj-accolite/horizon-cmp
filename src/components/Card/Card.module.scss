.card-container {
    transition: box-shadow 0.3s ease, border-color 0.3s ease;
  }
  
  .card {
    border: 1px solid grey;
    border-radius: 12px;
    overflow: hidden;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 16px;
    text-align: center;
  
    &:hover {
      box-shadow: 0 4px 16px rgba(0, 0, 0, 0.2);
      border-color: black;
    }
  
}