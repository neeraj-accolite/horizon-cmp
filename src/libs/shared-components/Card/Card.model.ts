export interface CardProps {
  children: React.ReactNode;
  style?: React.CSSProperties;
}
