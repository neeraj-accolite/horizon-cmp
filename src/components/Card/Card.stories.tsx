import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import { Card, CardProps } from './Card';

export default {
  title: 'Components/Card',
  component: Card,
  parameters: { layout: 'centered' },
} as Meta;

const Template: StoryFn<CardProps> = (args) => <Card {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'Generic Card',
  children: <div>Child Component</div>,
  width: '400px',
  height: '300px',
};
