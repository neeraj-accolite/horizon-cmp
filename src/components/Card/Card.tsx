import React from 'react';
import styles from './Card.module.scss';
import { CardProps } from './Card.model';

export const Card: React.FC<CardProps> = ({
  title,
  children,
  width,
  height,
}) => {
  return (
    <div className={`${styles['card-container']} transition-shadow`}>
      <div
        className={`${styles.card} flex flex-col items-center p-4 text-center`}
        style={{
          width: width,
          height: height,
        }}
      >
        <h2 className="text-xl mb-2">{title}</h2>
        <div className="flex-1 w-full">{children}</div>
      </div>
    </div>
  );
};

export type { CardProps };