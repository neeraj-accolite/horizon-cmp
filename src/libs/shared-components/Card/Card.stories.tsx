import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import { Card } from './Card';
import { CardProps } from './Card.model';

export default {
  title: 'Components/Card',
  component: Card,
  parameters: { layout: 'centered' },
} as Meta;

const Template: StoryFn<CardProps> = (args) => <Card {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: <div>Default Child Component</div>,
  style: {
    borderColor: '#D9D9D9',
    backgroundColor: '#F5F6F6',
    borderRadius: '8px',
    padding: '1rem',
  },
};

export const CustomStyle = Template.bind({});
CustomStyle.args = {
  children: <div>Custom Styled Child Component</div>,
  style: {
    borderColor: '#FF5733',
    backgroundColor: '#C70039',
    borderRadius: '16px',
    padding: '2rem',
    color: '#FFFFFF',
  },
};
