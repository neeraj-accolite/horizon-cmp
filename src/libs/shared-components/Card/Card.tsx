import React from 'react';
import { CardProps } from './Card.model';

const Card: React.FC<CardProps> = ({
  children,
  style = {},
  className = '',
}) => {
  return (
    <div style={{ ...style }} className={className}>
      {children}
    </div>
  );
};

export default Card;
