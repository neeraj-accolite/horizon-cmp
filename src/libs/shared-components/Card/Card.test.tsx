import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Card } from './Card';

describe('Card Component', () => {
  test('renders with default styles', () => {
    const { getByText } = render(
      <Card className="rounded-lg border border-gray-300 bg-gray-100 p-4">
        <div>Default Child Component</div>
      </Card>,
    );
    const childElement = getByText('Default Child Component');
    expect(childElement).toBeInTheDocument();
    expect(childElement.parentElement).toHaveClass(
      'border border-gray-300 bg-gray-100 rounded-lg p-4',
    );
  });

  test('renders with custom styles', () => {
    const { getByText } = render(
      <Card className="rounded-xl border border-red-500 bg-red-700 p-8 text-white">
        <div>Custom Styled Child Component</div>
      </Card>,
    );
    const childElement = getByText('Custom Styled Child Component');
    expect(childElement).toBeInTheDocument();
    expect(childElement.parentElement).toHaveClass(
      'border border-red-500 bg-red-700 rounded-xl p-8 text-white',
    );
  });

  test('renders children correctly', () => {
    const { getByText } = render(
      <Card>
        <div>Child Component</div>
      </Card>,
    );
    const childElement = getByText('Child Component');
    expect(childElement).toBeInTheDocument();
  });

  test('matches snapshot with default styles', () => {
    const { asFragment } = render(
      <Card className="rounded-lg border border-gray-300 bg-gray-100 p-4">
        <div>Default Child Component</div>
      </Card>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('matches snapshot with custom styles', () => {
    const { asFragment } = render(
      <Card className="rounded-xl border border-red-500 bg-red-700 p-8 text-white">
        <div>Custom Styled Child Component</div>
      </Card>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('matches snapshot with children only', () => {
    const { asFragment } = render(
      <Card>
        <div>Child Component</div>
      </Card>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
