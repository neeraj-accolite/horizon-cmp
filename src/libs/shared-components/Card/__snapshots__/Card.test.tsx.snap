// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot with children only 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div>
      Child Component
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card Component matches snapshot with custom styles 1`] = `
<DocumentFragment>
  <div
    class="rounded-xl border border-red-500 bg-red-700 p-8 text-white"
  >
    <div>
      Custom Styled Child Component
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card Component matches snapshot with default styles 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border border-gray-300 bg-gray-100 p-4"
  >
    <div>
      Default Child Component
    </div>
  </div>
</DocumentFragment>
`;
